client/core/trade_simnet_test.go

```diff
@@ -197,13 +196,13 @@ func startClients(ctx context.Context) error {
 	return nil
 }
 
-func setup() (context.CancelFunc, error) {
+func setup(t testing.TB) (context.CancelFunc, error) {
 	err := readWalletCfgsAndDexCert()
 	if err != nil {
 		return func() {}, fmt.Errorf("error reading wallet cfgs or dex cert (harnesses running?): %w", err)
 	}
 	ctx, cancelCtx := context.WithCancel(context.Background())
-	err = startClients(ctx)
+	err = startClients(t, ctx)
```

Comment:
Please see the `teardown` function about 7 lines down from here.   Those `...cfg.DBPath` fields are what we're now creating in `init` with the new `t.TempDir()`.  I believe what was removed in `TestMain` was really just for the temp _folder_ assuming each test cleaned up their actual DB file via `teardown`.  If each `t.TempDir` works as intended, `teardown` won't need those `os.RemoveAll` anymore.
I will test however.

Reply:
I have removed those `os.RemoveAll` lines in `teardown` function.

`t.TempDir` ensures that the temporary directory is removed when the test finishes. It will fail the test if it could not delete the directory https://github.com/golang/go/blob/0c6d8bb109adfa188190b2e6de59b88853ded378/src/testing/testing.go#L1099